crypto/http/http_client.c

```diff
@@ -285,10 +285,13 @@ static int set_content(OSSL_HTTP_REQ_CTX *rctx,
             && BIO_printf(rctx->mem, "Content-Type: %s\r\n", content_type) <= 0)
         return 0;
 
-    rctx->req = req;
-    if ((req_len = BIO_ctrl(req, BIO_CTRL_INFO, 0, NULL)) <= 0)
-        return 1; /* streaming BIO may not support querying size */
-    return BIO_printf(rctx->mem, "Content-Length: %ld\r\n", req_len) > 0;
+    /* streaming BIO may not support querying size */
+    if ((req_len = BIO_ctrl(req, BIO_CTRL_INFO, 0, NULL)) <= 0
+        || BIO_printf(rctx->mem, "Content-Length: %ld\r\n", req_len) > 0) {
+        rctx->req = req;
```

Comment:
Why did you move the assignment to here?
It is needed also in case `BIO_ctrl(req, BIO_CTRL_INFO, 0, NULL)` fails, for instance for a streaming BIO not support querying size.
Otherwise the request to be sent is omitted in such cases!

Reply:
Please read the code change again!

Reply:
Ah, you are right!
I did not notice the change of the logic, using `||`.